### PR TITLE
SDG Hello World Example

### PR DESCRIPTION
**Pull request by chrisalexiuk-nvidia**
## Description
Adds a simple "SDG tour" of the current features surrounding SDG in NeMo Curator. 

## Usage
Tutorial Content

## Checklist
- [x ] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb [hunk 1]
```diff
@@ -0,0 +1,1396 @@
...
+    "    base_url=\"https://integrate.api.nvidia.com/v1\",\n",
+    "    api_key=os.environ[\"NVIDIA_API_KEY\"],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we can initalize our NeMo Framework `OpenAIClient`!"
```
Comment (ryantwolf) [inline tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb:133]: Nit: "NeMo Framework" -> "NeMo Curator"

File: tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb [hunk 2]
```diff
@@ -0,0 +1,1396 @@
...
+   "source": [
+    "#### Modifying the Prompts\n",
+    "\n",
+    "Nemo Curator gives us granular control of each of the prompts at every step of each pipeline - let's look at how we can modify the prompts!\n",
+    "\n",
+    "We'll start with a simple example of modifying the prompt to another provided default.\n",
+    "\n",
+    "You can find all available pre-constructed prompts [here](https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/synthetic/prompts.py).\n",
+    "\n",
+    "> NOTE: It's important that when you're constructing new prompts you need to include the same placeholders (`{topic}`, `{n_openlines}`, etc.) to ensure smooth integration with `NemoGenerator`"
```
Comment (ryantwolf) [inline tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb:690]: `NemoGenerator` -> `NemotronGenerator`

File: tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb [hunk 3]
```diff
@@ -0,0 +1,1396 @@
...
+    "from nemo_curator.synthetic import NemotronGenerator\n",
+    "\n",
+    "generator = NemotronGenerator(curator_openai_client)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you'd like to skip forward to a specific pipeline, you can find them here:\n",
```
Comment (ryantwolf) [inline tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb:325]: Do these links work locally? On Github they do not

**Review by ryantwolf — CHANGES_REQUESTED** [review]
Thanks for the great work! It's super easy to follow. Do you mind also adding a section on the Math and Python entity classification modules?

File: tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb (hunk 3 above)
Comment (chrisalexiuk-nvidia) [reply tutorials/synthetic-data-hello-world/Synthetic Data Generation - Hello World Examples.ipynb:325]: They do work locally, yes.

**Comment by chrisalexiuk-nvidia** [comment]
Addressed issues. 

Added Math/Python Entity Classification.

**Review by ryantwolf — APPROVED** [review]
Looks great, thank you!

**Pull request merged**